src/header/common/link.rs

```diff
@@ -0,0 +1,975 @@
+    pub fn new(link_values: Vec<LinkValue>) -> Link {
+        Link { values: link_values }
+    }
+}
+
+#[allow(dead_code)]
+impl LinkValue {
+    /// Create LinkValue from URI-Reference
+    pub fn new(uri: &str) -> ::Result<LinkValue> {
+        match Uri::new(uri) {
+            Err(_) => Err(::Error::Header),
+            Ok(u) => Ok(
+                LinkValue {
+                    link: u,
+                    rel: None,
+                    anchor: None,
+                    rev: None,
+                    href_lang: None,
+                    media_desc: None,
+                    title: None,
+                    title_star: None,
+                    media_type: None,
+                    link_extension: None,
+                }
+            )
+        }
+    }
+
+    /// Get the LinkValue's value
+    pub fn link(&self) -> &Uri {
+        &self.link
+    }
+
+    /// Get the LinkValue's `rel` parameter
+    pub fn rel(&self) -> Option<&Vec<RelationType>> {
+        self.rel.as_ref()
+    }
+
+    /// Get the LinkValue's `anchor` parameter
+    pub fn anchor(&self) -> Option<&Uri> {
+        self.anchor.as_ref()
+    }
+
+    /// Get the LinkValue's `rev` parameter
+    pub fn rev(&self) -> Option<&Vec<RelationType>> {
+        self.rev.as_ref()
+    }
+
+    /// Get the LinkValue's `hreflang` parameter
+    pub fn href_lang(&self) -> Option<&Vec<LanguageTag>> {
+        self.href_lang.as_ref()
+    }
+
+    /// Get the LinkValue's `media` parameter
+    pub fn media_desc(&self) -> Option<&Vec<MediaDesc>> {
+        self.media_desc.as_ref()
+    }
+
+    /// Get the LinkValue's `title` parameter
+    pub fn title(&self) -> Option<&String> {
+        self.title.as_ref()
+    }
+
+    /// Get the LinkValue's `title*` parameter
+    pub fn title_star(&self) -> Option<&String> {
+        self.title_star.as_ref()
+    }
+
+    /// Get the LinkValue's `type` parameter
+    pub fn media_type(&self) -> Option<&Mime> {
+        self.media_type.as_ref()
+    }
+
+    /// Get the LinkValue's `link-extension` parameter
+    pub fn link_extension(&self) -> Option<&String> {
+        self.link_extension.as_ref()
+    }
+
+    /// Update LinkValue's `rel` parameter
+    pub fn push_rel(mut self, rel: RelationType) -> LinkValue {
+        let mut v = self.rel.take().unwrap_or(Vec::new());
+
+        v.push(rel);
+
+        self.rel = Some(v);
+
+        self
+    }
+
+    /// Set LinkValue's `anchor` parameter
+    pub fn set_anchor(mut self, anchor: &str) -> ::Result<LinkValue> {
+        match Uri::new(anchor) {
+            Err(_) => Err(::Error::Header),
+            Ok(uri) =>  {
+                self.anchor = Some(uri);
+
+                Ok(self)
+            }
+        }
+    }
+
+    /// Update LinkValue's `rev` parameter
+    pub fn push_rev(mut self, rev: RelationType) -> LinkValue {
+        let mut v = self.rev.take().unwrap_or(Vec::new());
+
+        v.push(rev);
+
+        self.rev = Some(v);
+
+        self
```

Comment:
No, `Option` just received a method to do this, but it's only in nightly, so this must do.